### PR TITLE
Submit functionality

### PR DESCRIPTION
**Pull request by sharonwarner**
When a user clicks on the 'Submit' button, mocked form data is saved to the database!

Screen Shot 2020-06-16 at 6 02 18 PM

There are a missing pieces of this PR: 1) unit tests, 2) proper authentication (right now a user will get a 500 if they try to save the form while unauthenticated. I'll add them as part of a separate PR so I can unblock folks waiting for this functionality.

===

Connects #286 

- [x] Snapshots updated, if output HTML/JS has changed (`npm run test:update-snapshots`)
- [x] Changes reviewed on mobile using devtools, if output HTML/CSS has changed
- [ ] Spanish checked by adding `?lng=es` to URL e.g. `/guide/benefits?lng=es`, if Spanish updated
- [ ] Automated tests added, if new functionality added
- [ ] Documentation (e.g. READMEs) updated, if relevant
- [ ] IE11 and Edge compatibility confirmed via manual testing in Browserstack, if new libraries added or newish JS/HTML/CSS features used for the first time in this codebase
- [ ] Accessibility & performance audit run using Google Lighthouse, if major changes made

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "jsdoc/require-returns": "warn",
     "jsdoc/require-returns-description": "off",
     "jsdoc/no-undefined-types": "off",
+    "jsdoc/check-tag-names": ["error", { "definedTags": ["jest-environment"] }],
```
Comment (sharonwarner) [inline .eslintrc:40]: This gets rid of those annoying github errors we were getting on PRs.

File: src/client/pages/RetroCertsCertificationPage/index.js [hunk 2]
```diff
@@ -71,30 +88,29 @@ function RetroCertsCertificationPage(props) {
...
-                        fromIndexToPathString(
-                          userData.weeksToCertify[weekForUser]
-                        )
-                  }
-                  type="submit"
-                >
-                  {weekForUser === numberOfWeeks ? (
-                    t("retrocerts-certification.button-submit")
-                  ) : (
+                {weekForUser === numberOfWeeks && (
```
Comment (kalvinwang) [inline src/client/pages/RetroCertsCertificationPage/index.js:91]: no need to change, just curious, is this functionally different or better than using `(weekForUser === numberOfWeeks) ? (Button) : (otherButton)`?

File: src/data/cosmos.js [hunk 3]
```diff
@@ -123,6 +123,14 @@ async function getFormDataByUserIdWithNewAuthToken(userId) {
   return item;
 }
 
+async function saveFormData(authToken, formData) {
+  const item = await getFormDataByAuthToken(authToken);
+  item.formData = formData;
+  item.confirmationNumber = uuidv4();
+  await insertForm(item);
```
Comment (kalvinwang) [inline src/data/cosmos.js:130]: oh haha i made insertForm as a placeholder, so you can just replace this line with `await insertItem(item, formsContainerName);` and delete the insertForm function below

File: src/client/pages/RetroCertsCertificationPage/index.js [hunk 4]
```diff
@@ -30,9 +32,24 @@ function RetroCertsCertificationPage(props) {
...
+        formData: mockedFormData,
+        authToken: sessionStorage.getItem(AUTH_STRINGS.authToken),
+      }),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        setUserData(data);
+        history.push(routes.retroCertsConfirmation);
+      })
+      .catch((error) => console.error(error));
```
Comment (idealisms) [inline src/client/pages/RetroCertsCertificationPage/index.js:51]: We need to handle errors, right? Like If the auth token is incorrect. Maybe add a TODO?

File: src/data/cosmos.js [hunk 5]
```diff
@@ -123,6 +123,14 @@ async function getFormDataByUserIdWithNewAuthToken(userId) {
   return item;
 }
 
+async function saveFormData(authToken, formData) {
+  const item = await getFormDataByAuthToken(authToken);
```
Comment (idealisms) [inline src/data/cosmos.js:127]: I think we need to handle if item is null (e.g., if the authToken is bad).

File: src/data/cosmos.js [hunk 6]
```diff
@@ -123,6 +123,14 @@ async function getFormDataByUserIdWithNewAuthToken(userId) {
...
+  item.formData = formData;
+  item.confirmationNumber = uuidv4();
```
Comment (idealisms) [inline src/data/cosmos.js:129]: Yeah, we need to decide what confirmation numbers look like. A uuid is maybe too confusing. Anyway, this is fine for now, but we should circle back with the designers & EDD.

File: src/client/pages/RetroCertsCertificationPage/index.js (hunk 2 above)
Comment (sharonwarner) [reply src/client/pages/RetroCertsCertificationPage/index.js:91]: I think it's more readable? But yeah, there's no functional difference.

File: src/client/pages/RetroCertsCertificationPage/index.js (hunk 4 above)
Comment (sharonwarner) [reply src/client/pages/RetroCertsCertificationPage/index.js:51]: Yep! There are a few auth / error handling errors that I need to update. I'll do that as part of a separate PR.

File: src/data/cosmos.js (hunk 3 above)
Comment (sharonwarner) [reply src/data/cosmos.js]: Oh lol. Updated!

**Pull request merged**